WcaOnRails/spec/controllers/registrations_controller_spec.rb

```diff
@@ -562,4 +562,105 @@
       expect(registrations.map(&:pos)).to eq [ 1 ]
     end
   end
+
+  describe 'POST #process_payment' do
+
+    context 'when not signed in' do
+      let(:competition) { FactoryGirl.create(:competition, :entry_fee, :visible, :registration_open, events: Event.where(id: %w(222 333))) }
+      sign_out
+
+      it 'redirects to the sign in page' do
+        post :process_payment, competition_id: competition.id
+        expect(response).to redirect_to new_user_session_path
+      end
+    end
+
+    context 'when signed in' do
+      let(:competition) { FactoryGirl.create(:competition, :entry_fee, :visible, :registration_open, events: Event.where(id: %w(222 333))) }
+      let!(:user) { FactoryGirl.create(:user, :wca_id) }
+      let!(:registration) { FactoryGirl.create(:registration, competition: competition, user: user) }
+
+      before :each do
+        sign_in user
+      end
+
+      it 'processes payment with valid credit card' do
+        Stripe.api_key = ENVied.STRIPE_API_KEY
+        token_id = Stripe::Token.create(
+          card: {
+            number: "4242424242424242",
+            exp_month: 12,
+            exp_year: 2017,
+            cvc: "314",
+          },
+        ).id
+        post :process_payment, competition_id: competition.id, stripeToken: token_id
+        expect(response).to redirect_to competition_register_path(competition)
+        expect(registration.reload.outstanding_entry_fees).to eq 0
```

Comment:
It's not super clear to me reading this test that `outstanding_entry_fees` actually changed here. Maybe it just started out as 0? To make this clearer, I'd vote for adding a `expect` before the POST that proves that `outstanding_entry_fees` is greater than zero before the payment.